### DISCUSSION TITLE
FetureEngineer and User_Defined_Features

### DISCUSSION THREAD

**Discussion by MaximCamilleri** [AI4Finance-Foundation/FinRL#763] [Q&A]
I was taking a look at the 'FeatureEngineer' function and noticed it has a parameter for user defined features. I was wandering if anyone could help me determine how I would add custom features to my environment (or maybe direct me to an example).

Specifically I am trying to use PCA on a set of technical indicators and then pass in the result as features.

**Comment by zhumingpassional** [comment]
you can calc your feature, and then append it after calculating finrl's indicators.
pls make sure that the dims are the same since finrl may delete NaN or None values.
